src/components/Header.js

```diff
@@ -41,7 +41,7 @@ class Header extends Component {
 
   shouldShowSearchBox = () => {
     const { searchPhrase, location } = this.props;
-    return location.pathname !== '/advance' && searchPhrase !== '';
+    return location.pathname == '/' && searchPhrase !== '';
```

Comment:
Using `==` will show warning in the console, can you change to `===` for stricter equality checking please?